File: Arrays and Linked List/Solutions/سلام به لیست پیوندی!/salam_to_ll.java
//Maryam Sadat Safavi
import java.util.Scanner;

public class LinkrdLst {
    public static void main(String[] args) {
        Scanner sc = new Scanner((System.in));
        LinkedList list = new LinkedList();
        while (true) {
            String input = sc.nextLine();
            if (input.split(" ")[0].equals("push")) {
                list.push(Integer.parseInt(input.split(" ")[1]));
            } else if (input.split(" ")[0].equals("pop")) {
                list.pop();
            } else if (input.split(" ")[0].equals("sort")) {
                list.sort();
            } else if (input.split(" ")[0].equals("print")) {
                list.print();
            } else if (input.split(" ")[0].equals("finish")) {
                System.exit(0);
            }
        }
    }
}

class LinkedList<Integer> {
    private Node head = null;
    private int size = 0;

    private class Node<Integer> {
        private int data;
        private Node<Integer> next;

        public Node(int d) {
            data = d;
            next = null;
        }

        public int getElement() {
            return this.data;
        }

        public Node<Integer> getNext() {
            return this.next;
        }

    }

    public void push(int x) {
        Node temp = new Node<>(0);
        temp.data = x;
        temp.next = head;
        head = temp;
        size++;
    }

    public void pop() {
        if (size != 0) {
            int elmnt = head.getElement();
            head = head.getNext();
            size--;
        } else {
            return;
        }
    }

    public void sort() {
        //bubble sort
        Node frst = head, scnd = null;
        int temp;
        if (head == null) {
            return;
        } else {
            while (frst != null) {
                scnd = frst.next;
                while (scnd != null) {
                    if (frst.data > scnd.data) {
                        temp = frst.data;
                        frst.data = scnd.data;
                        scnd.data = temp;
                    }
                    scnd = scnd.next;
                }
                frst = frst.next;
            }
        }
    }

    public void print() {
        Node currNode = this.head;
        int i = 0;
        if (size == 0) {
            System.out.println("list is empty");
        } else {
            while (i < size) {
                if (i == size - 1) {
                    System.out.print(currNode.data);
                } else {
                    System.out.print(currNode.data + " ");
                }
                currNode = currNode.next;
                i += 1;
            }
            System.out.println();
        }
    }
}

File: Arrays and Linked List/Solutions/سلام به لیست پیوندی!/salam_to_ll.py
#Lida Safari
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
class LinkedList:
    def __init__(self):
        self.head = None
        self.size = 0
    def push(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node
        self.size+=1
    def pop(self):
        if self.head == None:
            return
        else:
            self.head = self.head.next
            self.size -= 1
    def print(self):
        if self.head == None:
            print("list is empty")
            return
        tmp = self.head
        while tmp.next != None:
            print(tmp.data, end=" ")
            tmp = tmp.next
        print(tmp.data)
    def sort(self):
       # return
                for i in range(self.size):
                    tmp = self.head
                    while tmp.next != None:
                        if tmp.data > tmp.next.data:
                            tmp.data, tmp.next.data = tmp.next.data, tmp.data
                        tmp = tmp.next
my_linkedlist = LinkedList()
s = input().split()
order = s[0]
while order != "finish":
    if order == "push":
        my_linkedlist.push(int(s[1]))
    if order == "pop":
        my_linkedlist.pop()
    if order == "sort":
        my_linkedlist.sort()
    if order == "print":
        my_linkedlist.print()
    s = input().split()
    order = s[0]